csharp-advanced-topics/MutexClassInCSharp/MutexClassInCSharp/Program.cs

```diff
@@ -0,0 +1,16 @@
+﻿const string fileName = "numbers.txt";
+
+using Mutex mutex = new Mutex(initiallyOwned: false, "Global\\numbers_output", out bool mutexWasCreated);
+
+mutex.WaitOne();
+
+for (int number = 1; number <= 50; number++)
+{
+    File.AppendAllText(fileName, $"{number} ");
+    Thread.Sleep(100);
+}
+
+mutex.ReleaseMutex();
```

Comment:
This could be extracted into a new public static class with just this single method. We could also add a try/finally around the mutex set and release to ensure  releasing even in the presence of an exception. This would have the added benefit of removing the need for the empty class and the reflection call in the testing assembly. We could simply call the function from the test assembly. This main console then could still also call this code.

Just a thought. I might be missing something, but it seems to me that would be cleaner.